src/devices/wifi/nm-device-iwd.c

```diff
@@ -858,8 +884,14 @@ dbus_request_scan_cb (NMDevice *device,
 		}
 	}
 
-	g_dbus_proxy_call (priv->dbus_proxy, "Scan", g_variant_new ("()"),
-	                   G_DBUS_CALL_FLAGS_NONE, -1, NULL, NULL, NULL);
+	if (!priv->scanning && !priv->scan_requested) {
+		g_dbus_proxy_call (priv->dbus_proxy, "Scan",
+		                   g_variant_new ("()"),
+		                   G_DBUS_CALL_FLAGS_NONE, -1,
+		                   priv->cancellable, scan_cb, self);
+		priv->scan_requested = TRUE;
```

Comment:
maybe instead of having `priv->scan_requested`, use a separate `priv->cancellable_scan` instance? It feels confusing to me to reuse the same cancellable for multiple tasks.

Reply:
`priv->scan_requested` is used so that we avoid queuing a second scan even through `priv->scanning` is still FALSE while we're waiting for the scan to start.  So I don't think those two things are related.
I can add a `priv->cancellable_scan` but being new to glib I thought that was the whole benefit of using GCancellables.  For things like device shutdown where everything needs to be cancelled you don't have to keep one variable for each single task that the device may be doing, it seemed like it was made for this purpose.  The doc only says that it can be done but doesn't say if it's recommended or not.